src/normalize.js

```diff
@@ -26,6 +27,14 @@ const getVersionField = function ({ version }) {
   return version
 }
 
+// Array with all {node: ..., npm: ...} version pairs
+const getAllNodeNpmVersions = function (indexItems) {
+  return indexItems.map(({ version, npm }) => ({
+    node: version,
+    npm: npm || '0.0.0',
```

Comment:
The consumer will need to decide what to do when a specific Node.js release has no `npm` version. They could throw an error, print a message, ignore some logic, etc.
It would be simpler for the consumer to make that check with `npm !== undefined` instead of hardcoding `npm !== '0.0.0'`. From that angle, I believe it might be better to just leave `npm` `undefined` when it is not defined.
What do you think?

Reply:
On one hand - yes, I see your point. On the other hand, this will only be needed if the user cares about really ancient node versions, which probably nobody cares about, like `0.x`. And all other 99% of users will have to explicitly check that `npm !== undefined` if they use typescript because it'll be `string | undefined` instead of just `string`. And also it was easier to test this way :) But I'll probably agree that it's better to return `undefined`, because if I try `npm i -g npm@0.0.0` it'll throw "No matching version found" which isn't good, we could've prevented this...

Reply:
That's a great point you have about making it a "minimum" value, kindof like using `-Infinity` if it was a number. This could also comparison, i.e. no npm version is considered lower/older than any npm version. :thinking: 

The `npm` field was introduced in Node `0.6.3` with value `1.0.0-alpha.2`, and was `undefined` before that.

It is also possible to select different builds of Node.js by changing the `mirror` option: 
  - [Nightly builds](https://nodejs.org/download/nightly/index.json), [Release](https://unofficial-builds.nodejs.org/download/release/index.json), [Tests](https://nodejs.org/download/test/index.json) [V8 canary](https://nodejs.org/download/v8-canary/index.json), [ChakraCore](https://nodejs.org/download/chakracore-release/index.json): `npm` field is always defined, so no problem
  - RC (https://unofficial-builds.nodejs.org/download/rc/index.json): `npm` field is always undefined

That's a good point too about `string | undefined` requiring an additional `npm !== undefined` check for TypeScript users.

That being said, as you mention, `npm i npm@0.0.0` is not good either. 
This is actually the route that Node.js' `index.json` went too: they left the `npm` field undefined for older releases.

Since you might be the main user of this new field, do you feel like `0.0.0` would be simpler to use? If so, let's go ahead with it.

Reply:
Mm, interesting point about other mirrors, I didn't consider that.
`undefined` makes sense and it fine by me, I didn't even have to change the code, because I just check if npm version from engines section matches the default npm shipped with node:
```ts
export const isDefaultNpmVersion = async (node: string, npm: string) => {
  const nodeNormalized = await normalizeNodeVersion(node);
  const { versions } = await allNodeVersions();
  const defaultNpm = versions.find((x) => x.node === nodeNormalized)?.npm;
  return defaultNpm === npm;
};
```

Reply:
Alright, let's go with `undefined` then! 
Thanks for going thinking this detail through :)